test/cluster/test_tablets_lwt.py

```diff
@@ -576,3 +576,63 @@ async def test_lwt_coordinator_shard(manager: ManagerClient):
         matches = await n2_log.grep("CAS\\[0\\] successful")
         assert len(matches) == 1
         assert "shard 1" in matches[0][0]
+
+
+@pytest.mark.asyncio
+async def test_error_message_for_timeout_due_to_uncertainty(manager: ManagerClient):
+    # LWT can sometimes return WriteTimeout when it is uncertain whether the transaction
+    # was applied. In this case, the user should retry the transaction.
+    #
+    # Scenario:
+    # 1. The cluster has three nodes. The first node is configured to throw an error on accept,
+    #    and the second node is configured to wait for an explicit signal from the test.
+    # 2. Run the first LWT on the first node and wait until the coordinator processes the error from it.
+    # 3. Run another LWT, which should invalidate all promises made during the first one.
+    # 4. Signal the second node to proceed with accept; it must return a reject.
+    # 5. At this point, the coordinator is uncertain about the result of the first LWT. Any subsequent LWT
+    #    might either complete its Paxos round (if a quorum of promises observed the accept) or
+    #    overwrite it (if a quorum of promises never observed the accept).
+    #
+    # The test verifies that the WriteTimeout error returned by the system
+    # correctly reflects the cause of the problem.
+
+    logger.info("Bootstrapping cluster")
+    cmdline = [
+        '--logger-log-level', 'paxos=trace'
+    ]
+    servers = await manager.servers_add(3, cmdline=cmdline, auto_rack_dc="mydc")
+    (cql, hosts) = await manager.get_ready_cql(servers)
+
+    logger.info("Create a keyspace")
+    async with new_test_keyspace(manager, "WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 3} AND tablets = {'initial': 1}") as ks:
+        logger.info("Create a table")
+        await cql.run_async(f"CREATE TABLE {ks}.test (pk int PRIMARY KEY, c int);")
+
+        # accept on the first node returns an error
+        logger.info("Inject paxos_error_before_save_proposal")
+        await inject_error_one_shot_on(manager, "paxos_error_before_save_proposal", [servers[0]])
```

Comment:
This test uses error injections and it seems like it passed in CI in release mode. I don't understand how, something must have changed or CI is broken again and this test didn't even run.

Reply:
@yaronkaikov Why this test wasn't run in https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/18764/testReport?

I see this in the logs:
```
02:54:28.125 CRITICAL> Started /jenkins/workspace/scylla-master/scylla-ci/scylla/./test.py --list cluster/test_tablets_lwt::*
02:54:28.155 INFO> No boost found. Skipping pytest execution for boost tests.
```

Not sure if it's important or not.

Reply:
I think https://github.com/scylladb/scylla-pkg/pull/5345 fixed this issue. Let's see the next CI run.

Reply:
I see the test passed in CI this time